p5.html.markdown

```diff
@@ -10,44 +10,147 @@ filename: p5.js
 
+To run p5.js code, you can go to [the online editor](https://editor.p5js.org/).
+
 ```js
 ///////////////////////////////////
 // p5.js has two important functions to work with.
 
 function setup() {
-    // the setup function gets executed just once when the window is loaded
+  // the setup function gets executed just once when the window is loaded
 }
 function draw() {
-    // the draw function gets called every single frame. This means that for a frameRate(30) it would get called 30 times per second.
+  // the draw function gets called every single frame
+  // if the framerate is set to 30, it would get called 30 times every second
 }
 
 // the following code explains all features
 
 function setup() {
-    createCanvas(640, 480); // creates a new canvas element with 640px as width as 480px as height
-    background(128); // changes the background color of the canvas, can accept rgb values like background(100,200,20) else grayscale values like background(0) = black or background(255) = white
+  createCanvas(640, 480); // creates a new canvas element with 640px as width as 480px as height
+  background(128); // sets the background color to rgb(128, 128, 128)
+  // background('#aaf') // you can use hex codes and color names too
 }
 
 function draw() {
-    ellipse(10, 10, 50, 50); // creates a ellipse at the 10px from the left and 10px from the top with width and height as 50 each, so its basically a circle.
-    //remember in p5.js the origin is at the top-left corner of the canvas
+  background('#f2f2fc'); // usually, you call `background` in draw to clear the screen
+  // creates an ellipse at 10px from the top and 10px from the left, with width and height 37
+  ellipse(10, 10, 37, 37);
+  // remember in p5.js the origin is at the top-left corner of the canvas
+
+  if (mouseIsPressed) {
+    // mouseIsPressed is a boolean variable that is true when the mouse is down, and false otherwise
+
+    fill(0); // fill sets the fill color, which will stay until it is changed
+  } else {
+    fill(255, 255, 255, 240); // fill(a, b, c, d) sets the fill color to rgba(a, b, c, d)
+  }
+
+  ellipse(mouseX, mouseY, 80, 80);
+  // mouseX and mouseY are the x and y position of the mouse, respectively
+  // the above code creates and ellipse under the mouse, and fills it with white or black
+
+
+  // some other 2d primitives (shapes) you can draw:
+  rect(9, 3, 23, 26); // x, y, width, height
+  noFill(); // sets the fill color to transparent
+  triangle(100, 400, 130, 200, 200, 300); // x1, y1, x2, y2, x3, y3
+  point(100, 300); // create a point at x, y
+  // there are more, but they are more complex.
+}
 
-    if (mouseIsPressed) {
-        // mouseIsPressed is a boolean variable that changes to true if the mouse button is pressed down at that instant
+/** Bouncing balls animation
+ * See the finished product at
+ * https://editor.p5js.org/Kaamkiya/sketches/t5B78hIUg
```

Comment:
I don't like directly linking to a sketch because I never trust that these will stay up long enough.
Maybe better to just link to the editor, since the reader can just copy-paste the code anyway?

Reply:
Alright, sure. I changed the link and said that "you can copy-paste the code in to the editor", instead of the link to my personal sketch.